src/main/java/bio/terra/service/filedata/azure/util/BlobContainerCopier.java

```diff
@@ -0,0 +1,229 @@
+package bio.terra.service.filedata.azure.util;
+
+import com.azure.core.util.polling.SyncPoller;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobUrlParts;
+import com.azure.storage.blob.models.BlobCopyInfo;
+import com.azure.storage.blob.models.BlobListDetails;
+import com.azure.storage.blob.models.ListBlobsOptions;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BlobContainerCopier {
+
+  private static final int MIN_LIST_OPERATION_TIMEOUT_IN_SECONDS = 10;
+  private static final int MIN_POLLING_INTERVAL_IN_SECONDS = 2;
+  private static final Logger logger = LoggerFactory.getLogger(BlobContainerCopier.class);
+  private final BlobContainerClientFactory destinationClientFactory;
+
+  private String blobSourcePrefix = "";
+
+  private BlobContainerClientFactory sourceClientFactory;
+  private String sourceBlobUrl;
+  private String destinationBlobName;
+  private Duration listOperationTimeout = Duration.ofSeconds(MIN_LIST_OPERATION_TIMEOUT_IN_SECONDS);
+  private Duration pollingInterval = Duration.ofSeconds(MIN_POLLING_INTERVAL_IN_SECONDS);
+
+  private List<BlobCopySourceDestinationPair> sourceDestinationPairs;
+
+  public BlobContainerCopier(BlobContainerClientFactory destinationClientFactory) {
```

Comment:
I think I can make the package private. The tests for the class don't use the builder so I need to have the ability to create an instance directly.

Reply:
Is there any reason that we wouldn't just use the builder in the test?

Reply:
Using the builder in the test of this class, even as a connected test, feels incorrect to me. The scope of the test should be to the class and its creation and capabilities in isolation. The builder has its own test that makes sure that it wires up the copier for the appropriate cases.

Reply:
I think when I commented, I was thinking that the validation would mostly happen in the class. I tend to prefer lightweight builders and in that case it would fall into the pattern I'm proposing.  That being said, we could do this in a followon